spec/namespaces/flow_framework.yaml

```diff
@@ -0,0 +1,261 @@
+openapi: 3.1.0
+info:
+  title: OpenSearch Flow Framework API
+  description: OpenSearch Flow Framework API.
+  version: 1.0.0
+paths:
+  /_plugins/_flow_framework/workflow:
+    post:
+      operationId: flow_framework.create.0
+      x-operation-group: flow_framework.create
+      description: Create a workflow.
+      externalDocs:
+        url: https://opensearch.org/docs/latest/automating-configurations/api/create-workflow/
+      parameters:
+        - $ref: '#/components/parameters/flow_framework.create::query.provision'
+        - $ref: '#/components/parameters/flow_framework.create::query.validation'
+
+      requestBody:
+        $ref: '#/components/requestBodies/flow_framework.create'
+      responses:
+        '201':
+          $ref: '#/components/responses/flow_framework.create'
+  /_plugins/_flow_framework/workflow/{workflow_id}:
+    delete:
+      operationId: flow_framework.delete.0
+      x-operation-group: flow_framework.delete
+      description: Delete a workflow.
+      externalDocs:
+        url: https://opensearch.org/docs/latest/automating-configurations/api/delete-workflow/
+      parameters:
+        - $ref: '#/components/parameters/flow_framework.delete::query.clear_status'
+      responses:
+        '200':
+          $ref: '#/components/responses/flow_framework.delete'
+        '403':
+          description: if workflow id is null.
+    put:
+      operationId: flow_framework.update.0
+      x-operation-group: flow_framework.update
+      description: Update a workflow. You can only update a complete workflow if it has not yet been provisioned.
+      externalDocs:
+        url: https://opensearch.org/docs/latest/automating-configurations/api/create-workflow/
+      parameters:
+        - $ref: '#/components/parameters/flow_framework.update::query.update_fields'
+        - $ref: '#/components/parameters/flow_framework.get::path.workflow_id'
+      requestBody:
+        $ref: '#/components/requestBodies/flow_framework.update'
+      responses:
+        '201':
+          $ref: '#/components/responses/flow_framework.create'
+          description: It returns the workflow_id
```

Comment:
- 404 here if the previous workflow not found: https://github.com/opensearch-project/flow-framework/blob/b3f9d6514d2c0530c8bffac654fe175452673169/src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java#L416